$NetBSD: patch-ag,v 1.4 2020/03/26 02:35:31 joerg Exp $

- Fix build with png-1.5.
- Do not use void * as a synonym for gzFile, as gzFile is no longer
  void * in libz 1.2.6.
- Use const syntax correctly: const void * != void *const.

--- src/Util.cpp.orig	2004-05-20 17:42:37.000000000 +0000
+++ src/Util.cpp
@@ -47,7 +47,7 @@ extern "C" {
 #define _stricmp strcasecmp
 #endif // ! _MSC_VER
 
-static int (*utilGzWriteFunc)(gzFile, const voidp, unsigned int) = NULL;
+static int (*utilGzWriteFunc)(gzFile, voidpc, unsigned int) = NULL;
 static int (*utilGzReadFunc)(gzFile, voidp, unsigned int) = NULL;
 static int (*utilGzCloseFunc)(gzFile) = NULL;
 
@@ -79,7 +79,7 @@ bool utilWritePNGFile(const char *fileNa
     return false;
   }
 
-  if(setjmp(png_ptr->jmpbuf)) {
+  if(setjmp(png_jmpbuf(png_ptr))) {
     png_destroy_write_struct(&png_ptr,NULL);
     fclose(fp);
     return false;
@@ -478,7 +478,7 @@ bool utilIsGBAImage(const char * file)
 {
   cpuIsMultiBoot = false;
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gba") == 0)
@@ -502,7 +502,7 @@ bool utilIsGBAImage(const char * file)
 bool utilIsGBImage(const char * file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gb") == 0)
@@ -522,7 +522,7 @@ bool utilIsGBImage(const char * file)
 bool utilIsZipFile(const char *file)
 {
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".zip") == 0)
@@ -552,7 +552,7 @@ bool utilIsRarFile(const char *file)
 bool utilIsGzipFile(const char *file)
 {
   if(strlen(file) > 3) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       if(_stricmp(p, ".gz") == 0)
@@ -984,7 +984,7 @@ void utilWriteData(gzFile gzFile, variab
 
 gzFile utilGzOpen(const char *file, const char *mode)
 {
-  utilGzWriteFunc = (int (*)(void *,void * const, unsigned int))gzwrite;
+  utilGzWriteFunc = (int (*)(gzFile, const void *, unsigned int))gzwrite;
   utilGzReadFunc = gzread;
   utilGzCloseFunc = gzclose;
 
